### PR TITLE
dockerfile: separate toolchain for cargo audit

### LINKED ISSUE #2198: [devtool] Use separate toolchain as cargo audit dependency.
Currently we use the same toolchain to build cargo audit and Firecracker.  This indirectly prompts Firecracker toolchain updates whenever cargo audit db breaks compatibility and the new audit version depends on a newer toolchain.

### PR DESCRIPTION
**Pull request by aghecenco**
## Reason for This PR

Fixes #2198

## Description of Changes

Whenever the Rust advisory database format changes, `cargo-audit` needs to be upgraded in order to parse it, and sometimes that means upgrading the Rust toolchain. To avoid forcing a Firecracker-wide toolchain upgrade just to get a helper working, the toolchain versions are now explicitly separated.

This change will only take effect if it's agreed upon, and followed by a new container image.

## Testing

I tried this out with a diff that might occur in the future - newer toolchain for `cargo audit`:

```diff
diff --git a/tools/devctr/Dockerfile.x86_64 b/tools/devctr/Dockerfile.x86_64
index 425805f..33aed75 100644
--- a/tools/devctr/Dockerfile.x86_64
+++ b/tools/devctr/Dockerfile.x86_64
@@ -5,7 +5,7 @@ FROM ubuntu:18.04
 # dependencies layer intact for much longer.
 
 ARG RUST_TOOLCHAIN="1.43.1"
-ARG RUST_TOOLCHAIN_AUDIT="1.43.1"
+ARG RUST_TOOLCHAIN_AUDIT="1.44.1"
 ARG TINI_VERSION_TAG="v0.18.0"
 ARG TMP_BUILD_DIR=/tmp/build
 ARG FIRECRACKER_SRC_DIR="/firecracker"
```

Even though the newer toolchain is installed, the default toolchain remains the one Firecracker uses:

```bash
$ rustc --version
rustc 1.43.1 (8d69840ab 2020-05-04)
$ rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /usr/local/rust

installed toolchains
--------------------

1.43.1-x86_64-unknown-linux-gnu
1.44.1-x86_64-unknown-linux-gnu

installed targets for active toolchain
--------------------------------------

x86_64-unknown-linux-gnu
x86_64-unknown-linux-musl

active toolchain
----------------

1.43.1-x86_64-unknown-linux-gnu (default)
rustc 1.43.1 (8d69840ab 2020-05-04)
```

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

**Review by sandreim — APPROVED** [review]

**Comment by alindima** [comment]
The change looks good to me. However, I think we should include the commit with the new docker image in this PR.
Otherwise, there will be an undocumented discrepancy between the container version in devtool and the Dockerfiles.

File: tools/devctr/Dockerfile.x86_64 [hunk 1]
```diff
@@ -74,9 +75,10 @@ RUN mkdir "$TMP_BUILD_DIR" \
         && rustup target add x86_64-unknown-linux-musl \
         && rustup component add rustfmt \
         && rustup component add clippy-preview \
+        && rustup install "$RUST_TOOLCHAIN_AUDIT" \
         && cd "$TMP_BUILD_DIR" \
             && cargo install cargo-kcov \
-            && cargo install cargo-audit \
+            && cargo +"$RUST_TOOLCHAIN_AUDIT" install cargo-audit \
```
Comment (dianpopa) [inline tools/devctr/Dockerfile.x86_64:81]: Should we maybe use the latest toolchain available for cargo-audit, WDYT?

**Comment by acatangiu** [comment]
@aghecenco PR only needs a small change to use latest toolchain for cargo-audit, then it should be good to merge.

File: tools/devctr/Dockerfile.x86_64 (hunk 1 above)
Comment (aghecenco) [reply tools/devctr/Dockerfile.x86_64:83]: Updated to use v1.47.0.

Comment (acatangiu) [reply tools/devctr/Dockerfile.x86_64:83]: nit: I was thinking we could do:

```suggestion
            && cargo +stable install cargo-audit \
```

and ideally we could also do `rustup update` to get latest toolchain

**Review by acatangiu — APPROVED** [review]

File: tools/devctr/Dockerfile.x86_64 (hunk 1 above)
Comment (dianpopa) [reply tools/devctr/Dockerfile.x86_64:83]: Yup, this is what I was referring to when i suggested using the latest available toolchain.

**Review by dianpopa — APPROVED** [review]

**Comment by alindima** [comment]
> The change looks good to me. However, I think we should include the commit with the new docker image in this PR.
> Otherwise, there will be an undocumented discrepancy between the container version in devtool and the Dockerfiles.

Don't you think this is a good idea?

**Review by alindima — APPROVED** [review]

**Review by gc-plp — APPROVED** [review]

**Pull request merged**